pkg/hub/submarinerbrokerinfo/brokerinfo.go

```diff
@@ -360,19 +361,37 @@ func getTokenSecretForSA(ctx context.Context, client kubernetes.Interface, sa *c
 
 	for i := 0; i < len(saSecrets.Items); i++ {
 		if saSecrets.Items[i].Annotations[corev1.ServiceAccountNameKey] == sa.Name {
-			return &saSecrets.Items[i], nil
+			secret = &saSecrets.Items[i]
 		}
 	}
 
-	return nil, apierrors.NewNotFound(schema.GroupResource{
-		Group:    corev1.SchemeGroupVersion.Group,
-		Resource: "secrets",
-	}, sa.Name)
+	// Secret not found, so create one and return.
+	if secret == nil {
+		secret = &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:        GenerateBrokerName(sa.Name),
+				Annotations: map[string]string{corev1.ServiceAccountNameKey: sa.Name},
+			},
+			Type: corev1.SecretTypeServiceAccountToken,
+		}
+
+		secret, err = client.CoreV1().Secrets(sa.Namespace).Create(ctx, secret, metav1.CreateOptions{})
+		if err != nil && !apierrors.IsAlreadyExists(err) {
```

Comment:
If the result is an “already exists” error, this will end up returning an empty secret, AFAICT.

Reply:
Just realized that only reason we will hit this error is if someone created similar name secret manually with a different type. In that case we should just return with error and retry later. Earlier I was using a separate `returnSecret` but changed in refactoring which modfied the logic.